component/async/kafka/simple/duration_client.go

```diff
@@ -0,0 +1,130 @@
+package simple
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/beatlabs/patron/component/async/kafka"
+	"github.com/beatlabs/patron/log"
+)
+
+type durationClient struct {
+	client durationKafkaClientAPI
+}
+
+func newDurationClient(client durationKafkaClientAPI) (durationClient, error) {
+	if client == nil {
+		return durationClient{}, errors.New("empty client api")
+	}
+	return durationClient{client: client}, nil
+}
+
+func (d durationClient) getTimeBasedOffsetsPerPartition(ctx context.Context, topic string, since time.Time, timeExtractor kafka.TimeExtractor) (map[int32]int64, error) {
+	partitionIDs, err := d.client.getPartitionIDs(topic)
+	if err != nil {
+		return nil, err
+	}
+
+	responseCh := make(chan partitionOffsetResponse, len(partitionIDs))
+	d.triggerWorkers(ctx, topic, since, timeExtractor, partitionIDs, responseCh)
+	return d.aggregateResponses(ctx, partitionIDs, responseCh)
+}
+
+type partitionOffsetResponse struct {
+	partitionID int32
+	offset      int64
+	err         error
+}
+
+func (d durationClient) triggerWorkers(ctx context.Context, topic string, since time.Time, timeExtractor kafka.TimeExtractor, partitionIDs []int32, responseCh chan<- partitionOffsetResponse) {
+	for _, partitionID := range partitionIDs {
+		partitionID := partitionID
+		go func() {
+			offset, err := d.getTimeBasedOffset(ctx, topic, since, partitionID, timeExtractor)
+			select {
+			case <-ctx.Done():
+				return
+			case responseCh <- partitionOffsetResponse{
+				partitionID: partitionID,
+				offset:      offset,
+				err:         err,
+			}:
```

Comment:
am i missing something?
```suggestion
			}
```

Reply:
It's a select/case, it requires a `:`

Reply:
yeah, coreect. Didn't see it.